web/vibe/web/rest.d

```diff
@@ -1788,12 +1864,25 @@ private auto executeClientMethod(I, size_t ridx, ARGS...)
 		}
 	}
 
-	auto jret = request(URL(intf.baseURL), request_filter, request_body_filter,
+	headers["Accept"] = settings.content_type;
+	reqhdrs["Content-Type"] = null;
+	auto ret = request(URL(intf.baseURL), request_filter, request_body_filter,
 		sroute.method, url, headers, query.data, body_, reqhdrs, opthdrs,
 		intf.settings.httpClientSettings);
+	scope(exit) ret.dropBody();
+
+	static if (!is(RT == void)) {
+		string content_type = "";
+		if (const hdr = "Content-Type" in reqhdrs)
+			content_type = *hdr;
+		enum result_serializers = ResultSerializersT!(Func);
+		immutable serializer_ind = get_matching_content_type!(result_serializers)(content_type);
+		foreach (i, serializer; result_serializers)
+			if (serializer_ind == i)
+    			return serializer.deserialize(ret.bodyReader);
```

Comment:
```suggestion
    			{
				auto rng = streamInputRange(ret.bodyReader);
				return serializer.deserialize(rng);
			}
```
This should provide a range interface to the serializer to be consistent with `serialize`.

Reply:
> This should provide a range interface to the serializer to be consistent with `serialize`.

please see: https://github.com/vibe-d/vibe.d/pull/2492/files#r526105742

Reply:
> > This should provide a range interface to the serializer to be consistent with `serialize`.
> 
> please see: https://github.com/vibe-d/vibe.d/pull/2492/files#r526105742

@s-ludwig for the follow up PR, are you okay with my suggestion?
"I could create an input range from the stream by copying it, but this would be inefficient"